src/Confluent.SchemaRegistry.Serdes.Avro/SpecificSerializerImpl.cs

```diff
@@ -65,47 +104,62 @@ internal class SpecificSerializerImpl<T> : IAvroSerializerImpl<T>
             this.subjectNameStrategy = subjectNameStrategy;
 
             Type writerType = typeof(T);
+            if (writerType == typeof(ISpecificRecord))
+            {
+                _isMultipleSchemas = true;
+            }
+            else
+            {
+                schemaData.Add(writerType, ExtractSchemaData(writerType));
```

Comment:
i don't like using the dictionary in this way to only ever store a single value in this case, see previous suggestion.